lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -197,23 +197,23 @@ def is_node_up(self, host: Optional[str] = None) -> bool:
             return False
 
     def run_bin(self, bin_script_name: str, args: str = None, stdin: str = None) -> str:
-        """Run opensearch provided bin command, relative to OPENSEARCH_BIN.
+        """Run opensearch provided bin command, through the snap.
 
         Args:
             bin_script_name: opensearch script located in OPENSEARCH_BIN to be executed
             args: arguments passed to the script
             stdin: string input to be passed on the standard input of the subprocess.
         """
-        script_path = f"{self.paths.bin}/{bin_script_name}"
-        return self._run_cmd(script_path, args, stdin=stdin)
+        opensearch_command = f"opensearch.{bin_script_name}"
+        return self._run_cmd(opensearch_command, args, stdin=stdin)
 
     def run_script(self, script_name: str, args: str = None):
         """Run script provided by Opensearch in another directory, relative to OPENSEARCH_HOME."""
         script_path = f"{self.paths.home}/{script_name}"
         if not os.access(script_path, os.X_OK):
             self._run_cmd(f"chmod a+x {script_path}")
 
-        self._run_cmd(f"{script_path}", args)
+        self._run_cmd(f"snap run --shell opensearch.daemon -- {script_path}", args)
```

Comment:
nice, makes a lot of sense to run these scripts under the snap context. TBH we only use this logic [here](https://github.com/canonical/opensearch-operator/blob/44162927148aeb3032f9d6334f3a42731eb8d188/lib/charms/opensearch/v0/opensearch_base_charm.py#L1448).